tests/helpers_test.go

```diff
@@ -179,4 +185,80 @@ func uniqueObjects[T manifest.Object](t *testing.T, objects []T) []T {
 	return unique
 }
 
+func requireObjectsExists(t *testing.T, objects ...manifest.Object) {
+	t.Helper()
+	if !assertObjectsExists(t, objects...) {
+		t.FailNow()
+	}
+}
+
+func requireObjectsNotExists(t *testing.T, objects ...manifest.Object) {
+	t.Helper()
+	if !assertObjectsNotExists(t, objects...) {
+		t.FailNow()
+	}
+}
+
+func assertObjectsExists(t *testing.T, objects ...manifest.Object) bool {
+	t.Helper()
+	return assertObjectsExistsOrNot(t, objects, true)
+}
+
+func assertObjectsNotExists(t *testing.T, objects ...manifest.Object) bool {
+	t.Helper()
+	return assertObjectsExistsOrNot(t, objects, false)
+}
+
+type objectKindAndProject struct {
+	Kind    manifest.Kind
+	Project string
+}
+
+func (o objectKindAndProject) String() string {
+	if o.Project != "" {
+		return fmt.Sprintf("Kind: '%s' in Project: '%s'", o.Kind, o.Project)
+	}
+	return fmt.Sprintf("Kind: '%s'", o.Kind)
+}
+
+func assertObjectsExistsOrNot(t *testing.T, objects []manifest.Object, exists bool) bool {
+	t.Helper()
+	objectNamesPerKindAndProject := map[objectKindAndProject][]string{}
+	for _, object := range objects {
+		key := objectKindAndProject{
+			Kind: object.GetKind(),
+		}
+		if projectScoped, ok := object.(manifest.ProjectScopedObject); ok {
+			key.Project = projectScoped.GetProject()
+		}
+		v, ok := objectNamesPerKindAndProject[key]
+		if !ok {
+			v = make([]string, 0)
+		}
+		v = append(v, object.GetName())
+		objectNamesPerKindAndProject[key] = v
+	}
+
+	for key, names := range objectNamesPerKindAndProject {
+		headers := http.Header{}
+		if key.Project != "" {
+			headers.Set(sdk.HeaderProject, key.Project)
+		}
+		params := url.Values{v2.QueryKeyName: names}
+		objects, err := client.Objects().V1().Get(t.Context(), key.Kind, headers, params)
```

Comment:
Shouldn't we use V2 in the new code?

Reply:
that's a Get method, it's not in v2, I only moved Apply, Delete and DeleteByName there due to the breaking changes